addons/sale/models/sale_order.py

```diff
@@ -1245,6 +1245,26 @@ def get_portal_last_transaction(self):
         self.ensure_one()
         return self.transaction_ids._get_last()
 
+    def _get_order_lines_to_report(self):
+        down_payment_lines = self.order_line.filtered(lambda line:
+            line.is_downpayment
+            and not line.display_type
+            and not line._get_downpayment_state()
+        )
+
+        def show_line(line):
+            if not line.is_downpayment:
+                return True
+            elif line.display_type and down_payment_lines:
+                return True  # Only show the down payment section if down payments were posted
+            elif line in down_payment_lines:
+                return True  # Only show posted down payments
+            else:
+                return False
+
+        lines_to_report = self.order_line.filtered(show_line)
+        return lines_to_report, len(self) - len(lines_to_report)
```

Comment:
???

This new API is not clear at all, what is this second value?
It'll be always <= 0 since self is an unique order :thinking: